lib/plenario_web/controllers/api/detail_controller.ex

```diff
@@ -1,42 +1,64 @@
 defmodule PlenarioWeb.Api.DetailController do
   use PlenarioWeb, :api_controller
   import PlenarioWeb.Api.Utils, only: [render_page: 5, map_to_query: 2]
-  alias Plenario.Actions.MetaActions
   alias Plenario.{ModelRegistry, Repo}
+  alias Plenario.Actions.MetaActions
   alias PlenarioWeb.Controllers.Api.CaptureArgs
 
   defmodule CaptureColumnArgs do
     def init(opts), do: opts
 
     def call(conn, opts) do
-      columns = 
-        conn.params["slug"]
-        |> MetaActions.get()
-        |> MetaActions.get_column_names()
-      CaptureArgs.call(conn, opts ++ [fields: columns])
+      meta = MetaActions.get(conn.params["slug"], with_virtual_points: true) 
+      columns = MetaActions.get_column_names(meta)
+      vpfs = 
+        meta.virtual_points()
+        |> Enum.map(fn vpf -> vpf.name() end)
+      CaptureArgs.call(conn, opts ++ [fields: columns ++ vpfs])
+    end
+  end
+
+  defmodule CaptureBboxArg do
+    def init(opts), do: opts
+
+    def call(%Plug.Conn{params: %{"bbox" => geojson}} = conn, opts) do
+      meta = MetaActions.get(conn.params["slug"], with_virtual_points: true)
+      geom = Poison.decode!(geojson) |>  Geo.JSON.decode()
+      geom = %{geom | srid: 4326}
+
+      vpf_query_map = 
+        meta.virtual_points()
+        |> Stream.map(fn vpf -> vpf.name() end)
+        |> Enum.map(fn vpf_name -> {vpf_name, {"in", geom}} end)
+
+      assign(conn, opts[:assign], vpf_query_map)
+    end
+
+    def call(conn, opts) do
+      assign(conn, opts[:assign], [])
     end
```

Comment:
This plug handles generating individual geospatial queries for each virtual point field for a dataset.